crates/transpiler/src/passes/litinski_transformation.rs

```
@@ -0,0 +1,213 @@
+pub fn extract_rotations(circuit: &[(String, Vec<usize>)], nqubits: usize) -> Vec<(bool, String)> {
+    let mut clifford = Clifford::identity(nqubits);
+    let mut rotations: Vec<(bool, String)> = Vec::new();
+
+    for (gate_name, qbits) in circuit.iter() {
+        match gate_name.as_str() {
+            "id" => {}
+            "x" => clifford.append_x(qbits[0]),
+            "y" => clifford.append_y(qbits[0]),
+            "z" => clifford.append_z(qbits[0]),
+            "h" => clifford.append_h(qbits[0]),
+            "s" => clifford.append_s(qbits[0]),
+            "sdg" => clifford.append_sdg(qbits[0]),
+            "sx" => clifford.append_sx(qbits[0]),
+            "sxdg" => clifford.append_sxdg(qbits[0]),
+            "cx" => clifford.append_cx(qbits[0], qbits[1]),
+            "cz" => clifford.append_cz(qbits[0], qbits[1]),
+            "cy" => clifford.append_cy(qbits[0], qbits[1]),
+            "swap" => clifford.append_swap(qbits[0], qbits[1]),
+            "iswap" => clifford.append_iswap(qbits[0], qbits[1]),
+            "ecr" => clifford.append_ecr(qbits[0], qbits[1]),
+            "dcx" => clifford.append_dcx(qbits[0], qbits[1]),
+            "rz" => {
+                rotations.push(clifford.get_inverse_z(qbits[0]));
+            }
+            _ => panic!("Unsupported gate {}", gate_name),
+        }
+    }
+
+    rotations
+}
+
+#[pyfunction]
+#[pyo3(name = "run")]
+pub fn run_litinski_transformation(
+    py: Python,
+    dag: &mut DAGCircuit,
+) -> PyResult<Option<DAGCircuit>> {
+    let op_counts = dag.get_op_counts();
+
+    // Skip the pass if there are no rotation gates.
+    if op_counts
+        .keys()
+        .all(|k| !ROTATION_GATE_NAMES.contains(&k.as_str()))
+    {
+        return Ok(None);
+    }
+
+    // Skip the pass if there are unsupported gates.
+    if !op_counts
+        .keys()
+        .all(|k| SUPPORTED_GATE_NAMES.contains(&k.as_str()))
+    {
+        return Ok(None);
```

I feel like this should return an error saying that the pass cannot be run due to gates XYZ that are not supported, what do you think?

---

Returning an error makes sense: done in [5aab4b6](https://github.com/Qiskit/qiskit/pull/14753/commits/5aab4b66398cd175b8657c446ac04e5f6cbf3e00).